tests/RobotFramework/tests/pkcs11/compatibility.robot

```diff
@@ -190,3 +190,72 @@ Custom Setup
     Register Device With Cumulocity CA    ${DEVICE_SN}    csr_path=${csr_path}
 
     Unset tedge-p11-server Uri
+
+Connect to C8y using new ECDSA keypair
+    [Documentation]    Connects to C8y with a newly generated keypair and a self-signed certificate.
+    ...    The private key is saved on the token, and the self-signed certificate is registered with c8y.
+    [Arguments]    ${curve}=secp256r1
+    # if type == ECDSA, curve of the key - one of {secp256r1, secp384r1, secp521r1}
+
+    # We could alternatively use Cumulocity CA to start with a signed cert, but for testing certificate renewal, we want
+    # to test both renewing a self-signed cert and a cert issued by C8y CA. When we start with self-signed cert, after
+    # the first renewal we get a cert signed by CA, so we test all scenarios by just doing renew 2 times.
+    ${label}=    Set up new PKCS11 ECDSA keypair    curve=${curve}
+
+    ${cert_path}=    Set Variable    /etc/tedge/device-certs/${label}.pem
+    Execute Command    cmd=tedge config set device.cert_path ${cert_path}
+
+    Execute Command
+    ...    cmd=printf 'dn = "CN=${DEVICE_SN}, OU=Test Device, O=Thin Edge, ST=QLD, C=AU"\nca\n' | GNUTLS_PIN=123456 certtool --generate-self-signed --template /dev/stdin --outfile "${cert_path}" --load-privkey "pkcs11:token=tedge;object=${label}"
+    # Execute Command    cmd=tedge config set device.cryptoki.uri "pkcs11:token=tedge;object=${label}"
+    Set tedge-p11-server Uri    value=pkcs11:token=tedge;object=${label}
+
+    Execute Command
+    ...    cmd=sudo env C8Y_USER="${C8Y_CONFIG.username}" C8Y_PASSWORD="${C8Y_CONFIG.password}" tedge cert upload c8y
+    ThinEdgeIO.Register Certificate For Cleanup
+
+    Tedge Reconnect Should Succeed
+
+Connect to C8y using new RSA keypair
+    [Documentation]    Connects to C8y with a newly generated keypair and a self-signed certificate.
+    ...    The private key is saved on the token, and the self-signed certificate is registered with c8y.
+    [Arguments]    ${bits}=4096    # length in bits of the RSA key - one of {1024, 2048, 3072, 4096}
+
+    # We could alternatively use Cumulocity CA to start with a signed cert, but for testing certificate renewal, we want
+    # to test both renewing a self-signed cert and a cert issued by C8y CA. When we start with self-signed cert, after
+    # the first renewal we get a cert signed by CA, so we test all scenarios by just doing renew 2 times.
+
+    # guarantee name of the object is unique even if multiple keys of the same type and bits/curve are generated
```

Comment:
todo: Comment out of place (should be next to generating random string)

Reply:
I removed the comment as it wasn't providing much value for the compatibility tests as these tests are meant to be self-contained and won't be extended much in the future (apart from the future planned improvements to the keywords surrounding certificate handling).  Resolved in c9ea05c9885f5931fc0513e18fa9491f4cab8b5b